ComponentKit/Core/CKBuildComponent.mm

```diff
@@ -0,0 +1,63 @@
+//
+//  CKBuildComponent.m
+//  ComponentKit
+//
+//  Created by Oliver Rickard on 3/28/17.
+//
+//
+
+#import "CKBuildComponent.h"
+
+#import "CKComponentBoundsAnimation.h"
+#import "CKComponentBoundsAnimationPredicates.h"
+#import "CKComponentInternal.h"
+#import "CKComponentScopeRoot.h"
+#import "CKComponentSubclass.h"
+#import "CKThreadLocalComponentScope.h"
+
+static CKComponentBoundsAnimation boundsAnimationFromPreviousScopeRoot(CKComponentScopeRoot *newRoot, CKComponentScopeRoot *previousRoot)
+{
+  NSMapTable *scopeFrameTokenToOldComponent = [NSMapTable strongToStrongObjectsMapTable];
+  [previousRoot
+   enumerateComponentsMatchingPredicate:&CKComponentBoundsAnimationPredicate
+   block:^(id<CKScopedComponent> component) {
+     CKComponent *oldComponent = (CKComponent *)component;
+     id scopeFrameToken = [oldComponent scopeFrameToken];
+     if (scopeFrameToken) {
+       [scopeFrameTokenToOldComponent setObject:oldComponent forKey:scopeFrameToken];
+     }
+   }];
+  
+  __block CKComponentBoundsAnimation boundsAnimation {};
+  [newRoot
+   enumerateComponentsMatchingPredicate:&CKComponentBoundsAnimationPredicate
```

Comment:
Before, all of this was implemented inside the scope root itself. Now it's implemented in a few functions above the scope root using a generic API.